### DISCUSSION TITLE
Return only a max limit of trees in Spanning Tree Iterator

### DISCUSSION THREAD

**Discussion by shariharan2002** [networkx/networkx#5956] [Q&A] [answered]
Hi,
I am heavily using Spanning Tree Iterator function as part of the NetworkX package. Since I am working on large graphs there are many combinations of spanning trees possible.
The thing is, for accessing each spanning tree I use the below statement
spanning_tree_list=list(nx.SpanningTreeIterator(G))
nx.draw(spanning_tree_list[0])  # or any index i

Now, since I use the function heavily I don't want it to return all possible combiantions. It becomes a huge bottleneck. Instead I want it to return only max 10 or 20 trees. That s' all.
the below is still causing networkx to calculate all trees and then return 10 or 20, and therefore its still a bottleneck
spanning_tree_list=list(nx.SpanningTreeIterator(G))[0:10]

How do I resolve this issue?

Thanks

**Comment by MridulS** [comment, marked as answer]
By doing `spanning_tree_list=list(nx.SpanningTreeIterator(G))[0:10]` you are calculating all the Spanning Trees by doing the cast to `list()`. To slice an iterator you can use `itertools.islice()`.

To get the first 10 spanning trees from this iterator you can do something like:
```python
>>> import itertools
>>> spanning_tree_list = list(itertools.islice(nx.SpanningTreeIterator(G), 10))
```

**Reply by shariharan2002** [reply]
Thanks a lot. You have resolved the bottleneck. Forgive me for my Python illiteracy. Hope I have not been of much nuisance
Thanks a lot again.
